lib/LedgerSMB/Admin/Command/destroy.pm

```diff
@@ -0,0 +1,177 @@
+
+package LedgerSMB::Admin::Command::destroy;
+
+=head1 NAME
+
+LedgerSMB::Admin::Command::destroy - ledgersmb-admin 'destroy' command
+
+=cut
+
+use strict;
+use warnings;
+
+use Syntax::Keyword::Try;
+
+use LedgerSMB::Admin::Command;
+use LedgerSMB::Database;
+
+use Moose;
+extends 'LedgerSMB::Admin::Command';
+use namespace::autoclean;
+
+my $logger;
+
+sub run {
+    my ($self, $dbname) = @_;
+    $logger = $self->logger;
+    my $existing_db = $self->connect_data_from_arg($dbname);
+    my $connect_data = {
+        $self->config->get('connect_data')->%*,
+        $existing_db->%*,
+    };
+    my $db = LedgerSMB::Database->new(
+        connect_data => $connect_data,
+        );
+    my $role_prefix = _get_role_prefix($db);
+
+    $db->drop
+        or die $db->errstr;
+    $logger->info('Database successfully destroyed');
+
+    my $connect_admin = {
+        $self->config->get('connect_data')->%*,
+        $existing_db->%*,
+        dbname => ($self->config->get('admindb') // 'postgres'),
+    };
+    $db = LedgerSMB::Database->new(
+            connect_data => $connect_admin,
+        );
+    my $dbh = $db->connect;
+
+    # Scan databases which could be using the same profile
+    my $sth =
+        $dbh->prepare('SELECT datname FROM pg_database WHERE NOT datistemplate');
+
+    $sth->execute
+        or die $dbh->errstr;
+
+    my @lsmb_databases;
+    while (my $database = $sth->fetchrow) {
+        push @lsmb_databases, $database;
+    }
+    $sth->finish;
+    $dbh->disconnect;
+
+    my $role_prefix_usage = 0;
+    for my $database (@lsmb_databases) {
+        try {
+            my $connect_database = {
+                $self->config->get('connect_data')->%*,
+                $existing_db->%*,
+                dbname => $database,
+            };
+            my $db = LedgerSMB::Database->new(
+                    connect_data => $connect_database,
+                );
+            my $database_role_prefix = _get_role_prefix($db);
+            if ( defined $database_role_prefix && $role_prefix eq $database_role_prefix ) {
+                $role_prefix_usage++;
+                $logger->debug("'$role_prefix' used by $database");
+            }
+        }
+        catch ($e) {
+            $logger->error("ERROR: $e");
```

Comment:
We need to prefix the error message with "ERROR:" even when already using the `error` logger???

Reply:
Redundant. Fixed